src/ripple/app/main/LoadManager.cpp

```diff
@@ -129,23 +129,32 @@ void LoadManager::run ()
             auto const reportingIntervalSeconds = 10s;
             if (armed && (timeSpentDeadlocked >= reportingIntervalSeconds))
             {
+                constexpr auto deadlockTimeLimit = 90s;
+
                 // Report the deadlocked condition every 10 seconds
                 if ((timeSpentDeadlocked % reportingIntervalSeconds) == 0s)
                 {
-                    JLOG(journal_.warn())
-                        << "Server stalled for "
-                        << timeSpentDeadlocked.count() << " seconds.";
+                    if (timeSpentDeadlocked < deadlockTimeLimit)
+                    {
+                        JLOG(journal_.warn())
+                            << "Server stalled for "
+                            << timeSpentDeadlocked.count() << " seconds.";
+                    }
+                    else
+                    {
+                        JLOG(journal_.fatal())
+                            << "Deadlock detected. Deadlocked time: "
+                            << timeSpentDeadlocked.count()
+                            << "s JobQueue Overloaded: "
+                            << app_.getJobQueue().isOverloaded();
+                    }
                 }
 
-                // If we go over 90 seconds spent deadlocked, it means that
+                // If we go over the deadlockTimeLimit spent deadlocked, it means that
```

Comment:
Nit: Is the comment better placed at the fatal log?

Reply:
We may also want to move the assert to line 145 eg. `assert(false)`

Reply:
@miguelportilla I think I'd rather keep the assert outside the reporting interval. But I don't really object if you want to move it.

Reply:
@seelabs I don't feel strongly about either one and okay as is.